### PR TITLE
Updates to v0.36 release notes

### PR DESCRIPTION
**Pull request by trbromley**
**Context:**

The release notes need to be tidied in time for the 0.36 release on May 7th.

**Description of the Change:**

This is the first phase of changes to the release notes. In this PR we add categories to major features and improvements and categorize the existing PRs.

**Recommendation for reviewers:**

- It is important to merge this first changelog update PR relatively quickly as it often goes out of date with `master` and has merge conflicts. Hence, we typically do not worry about the changelog entries themselves, just that the categories are added and we believe entries are in the right category. Changes to the entries themselves, and writing polished summaries of major features, will happen in a subsequent PR.
- There are so many changes that the diff can be hard to follow. It is easiest to simply view the docs build to check if you are happy.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -118,132 +163,98 @@
...
-* `qml.transforms.broadcast_expand` now supports shot vectors when returning `qml.sample()`.
-  [(#5473)](https://github.com/PennyLaneAI/pennylane/pull/5473)
+  Note that the grouping indices refer to the lists returned by `Sum.terms()`, not `Sum.operands`.
 
-* `LightningVJPs` is now compatible with Lightning devices using the new device API.
-  [(#5469)](https://github.com/PennyLaneAI/pennylane/pull/5469)
+* Added new function `qml.operation.convert_to_legacy_H` to convert `Sum`, `SProd`, and `Prod` to `Hamiltonian` instances.
+  [(#5309)](https://github.com/PennyLaneAI/pennylane/pull/5309)
 
 * The `qml.is_commuting` function now accepts `Sum`, `SProd`, and `Prod` instances.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:249]: Also might not be bad here to have a code example

**Review by isaacdevlugt — COMMENTED** [review]
A decent amount of entries that could have quick & easy code examples IMO. Thoughts? Happy to add them where applicable :)

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -118,132 +163,98 @@
    -1.0 * Y(1) @ Y(0)]
   ```
 
-<h3>Improvements 🛠</h3>
+<h4>Simulate mixed-state qutrit systems 3️⃣</h4>
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/releases/changelog-dev.md:166]: It was, but I think it's worth giving this an extra highlight (we can also give a shout out in the text)

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -118,132 +163,98 @@
...
-  the operator it wraps.
-  [(#5473)](https://github.com/PennyLaneAI/pennylane/pull/5473)
 
-* `qml.transforms.broadcast_expand` now supports shot vectors when returning `qml.sample()`.
-  [(#5473)](https://github.com/PennyLaneAI/pennylane/pull/5473)
+  Note that the grouping indices refer to the lists returned by `Sum.terms()`, not `Sum.operands`.
 
-* `LightningVJPs` is now compatible with Lightning devices using the new device API.
-  [(#5469)](https://github.com/PennyLaneAI/pennylane/pull/5469)
+* Added new function `qml.operation.convert_to_legacy_H` to convert `Sum`, `SProd`, and `Prod` to `Hamiltonian` instances.
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/releases/changelog-dev.md:244]: Yes this can be done as we deprecate the old op arithmetic style (probably more towards Q3 or Q4 for a bigger thing like this)

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -266,75 +274,77 @@
...
+<h4>Other improvements</h4>
+
+* `qml.ops.Conditional` now stores the `data`, `num_params`, and `ndim_param` attributes of
+  the operator it wraps.
+  [(#5473)](https://github.com/PennyLaneAI/pennylane/pull/5473)
+
+* The `molecular_hamiltonian` function calls `PySCF` directly when `method='pyscf'` is selected.
+  [(#5118)](https://github.com/PennyLaneAI/pennylane/pull/5118)
+
+* Upgraded `null.qubit` to the new device API. Also, added support for all measurements and various modes of differentiation.
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/releases/changelog-dev.md:323]: It's also intended for use in resource estimation: https://docs.pennylane.ai/en/stable/code/qml_resource.html#tracking-resources-for-custom-operations
But we don't have enough content to make that clear. A demo is upcoming on resource estimation with errors though (need to check if that will use `null.qubit`).

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -4,74 +4,119 @@
 
 <h3>New features since last release</h3>
 
-* Added a qml.capture module that will contain PennyLane's own capturing mechanism for hybrid
-  quantum-classical programs.
-  [(#5509)](https://github.com/PennyLaneAI/pennylane/pull/5509)
+<h4>Estimate errors in a quantum circuit 🧮</h4>
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/releases/changelog-dev.md:7]: I hope the content of this section will be completely reworked in a follow up PR :+1:

File: doc/releases/changelog-dev.md [hunk 6]
```diff
@@ -118,132 +163,98 @@
...
-  [(#5159)](https://github.com/PennyLaneAI/pennylane/pull/5159)
+ * Created the `DefaultQutritMixed` class, which inherits from `qml.devices.Device`, with an implementation 
+  for `preprocess`.
+  [(#5451)](https://github.com/PennyLaneAI/pennylane/pull/5451)
 
-  ```python
-  @qml.prod
-  def generator(wires):
-        qml.Hadamard(wires=wires)
+<h4>Work easily and efficiently with operators 🔧</h4>
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/releases/changelog-dev.md:181]: Yes, we need to add some details to let users know that op arithmetic has been updated, pointing to the quickstart, etc.

**Review by isaacdevlugt — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 7]
```diff
@@ -4,74 +4,117 @@
...
+  [(#5464)](https://github.com/PennyLaneAI/pennylane/pull/5464)
+  [(#5465)](https://github.com/PennyLaneAI/pennylane/pull/5465)
+
+
+<h4>Access an extended arsenal of quantum algorithms 🏹</h4>
+
+* The `FABLE` template is added for efficient block encoding of matrices. Users can now call FABLE to efficiently construct circuits according to a user-set approximation level. 
+  [(#5107)](https://github.com/PennyLaneAI/pennylane/pull/5107)
+
+* Create the `qml.Reflection` operator, useful for amplitude amplification and its variants.
```
Comment (DSGuala) [inline doc/releases/changelog-dev.md:38]: I would put `qml.AmplitudeAmplification` template above `qml.Reflection`. Not a strong opinion.

**Review by DSGuala — APPROVED** [review]
Looks good to me. Couldn't find any typos or mistakes. Left two small comments, no blockers from my side.
